common/types_test.go

```diff
@@ -0,0 +1,31 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBigInt(t *testing.T) {
+	var v BigInt
+
+	textRef := []byte("11111111111111111111")
+	err := v.UnmarshalText(textRef)
```

Comment:
is there a slightly more end-to-end way to exercise MarshalText/UnmarshalText? would better to do that so we know the pointerness of the receiver is right

Reply:
I searched around and it looks like the only prominent use of MarshalText is by `encoding/json` for objects that don't define `MarhsalJSON`. So:
- Maybe technically we could omit (Un)marshalJSON and retain the same behavior
- Or we could keep MarshalJSON and remove MarshalText because the latter doesn't get used much in isolation

Anyway. I'd call the current state of affairs Good Enough. Thank you for the tests.

Reply:
ok I've found that we can't omit marshal/unmarshal json. we need these to override big.Int's marshal/unmarshal json, which use json numbers instead of strings

proceeding with removing our marhsal/unmarshal text. we're still able to call the embeded big.Int's implementation. I'm leaving the test in.